### PR TITLE
Move a part of the Ingest workflow into new Step Function

### PR DESCRIPTION
**Pull request by techncl**

### CODE REVIEW

File: templates/sfn/ingest_sfn_definition.json.tpl [hunk 1]
```diff
@@ -344,102 +316,17 @@
...
-          "ErrorEquals": [
-            "States.ALL"
-          ],
-          "IntervalSeconds": 2,
-          "MaxAttempts": 6,
-          "BackoffRate": 2
+        "StateMachineArn": "arn:aws:states:eu-west-2:${account_id}:stateMachine:${ingest_run_workflow_sfn_name}",
+        "Input": {
+          "StatePayload": "$",
+          "AWS_STEP_FUNCTIONS_STARTED_BY_EXECUTION_ID.$": "$$.Execution.Id"
```
Comment (kurtismash) [inline templates/sfn/ingest_sfn_definition.json.tpl:326]: Didn't know about this, nice find!

**Review by kurtismash — COMMENTED** [review]
Almost there, just a few comments. I don't have the mental capacity to visualise the Step Function by reading the definition so have skimmed over most of the JSON files.

File: common.tf [hunk 2]
```diff
@@ -34,6 +35,7 @@ locals {
   sse_encryption                                       = "sse"
   visibility_timeout                                   = 180
   redrive_maximum_receives                             = 5
+  ingest_run_workflow_sfn_arn                          = "arn:aws:states:eu-west-2:${data.aws_caller_identity.current.account_id}:stateMachine:${local.ingest_run_workflow_step_function_name}"
```
[the remark this answers is not in the document]
Comment (techncl) [reply common.tf:38]: We do this throughout this terraform repo so this is probably a question for @MancunianSam

File: custodial_copy_queue_creator.tf
[the remark this answers is not in the document]
Comment (techncl) [reply custodial_copy_queue_creator.tf:1]: That's weird; I'm not even sure how this happened.

File: common.tf [hunk 3]
```diff
@@ -265,6 +265,21 @@ module "dr2_ingest_step_function" {
   }
 }
 
+module "dr2_ingest_run_workflow_step_function" {
+  source = "git::https://github.com/nationalarchives/da-terraform-modules//sfn"
+  step_function_definition = templatefile("${path.module}/templates/sfn/ingest_run_workflow_sfn_definition.json.tpl", {
+    step_function_name                        = local.ingest_run_workflow_step_function_name
+    account_id                                = var.account_number
```
[the remark this answers is not in the document]
Comment (techncl) [reply common.tf:272]: I'm not sure; is it possible that a caller doesn't have the same account id as the thing it is calling? @MancunianSam, do you know?

File: common.tf (hunk 2 above)
[the remark this answers is not in the document]
Comment (MancunianSam) [reply common.tf:38]: We should but as Nigel says, it's everywhere. I'll put a ticket in to make it a parameter (even though that's the only region that we use) as it's going to need changes in a lot of places. This is fine for now though.

File: common.tf (hunk 3 above)
[the remark this answers is not in the document]
Comment (MancunianSam) [reply common.tf:272]: It's a good point. I think for the most part `var.account_number` would do it. Again, it's everywhere so leave this and I'll put a ticket in to settle on one or the other

**Review by kurtismash — APPROVED** [review]

**Review by kurtismash — APPROVED** [review]

**Pull request merged**